xcodeproj/internal/non_xcode_targets.bzl

```diff
@@ -98,17 +99,12 @@ rules_xcodeproj requires {} to have `{}` set.
         transitive_infos = transitive_infos,
     )
 
-    mergable_xcode_library_targets = [
-        struct(
-            id = target.id,
-            product_path = target.product.file_path,
-        )
-        for target, providers in [
-            (info.xcode_target, info.compilation_providers)
+    mergable_xcode_library_targets = memory_efficient_depset(
+        transitive = [
+            info.mergable_xcode_library_targets
             for info in transitive_infos
         ]
-        if providers._is_xcode_library_target
-    ]
+    )
```

Comment:
@mattrobmattrob Does this change make sense to you?

Reply:
Seems reasonable (and better) to me especially given no fixture changes WRT merging. 👍